### ISSUE TITLE
[Regression] In Castlevania: Harmony of Despair (NPEB00563, NPUB30505) is crashing after playing one game

### ISSUE THREAD

**Issue by RocherBrockas** [RPCS3/rpcs3#16458]
### Quick summary

This game use to work perfectly in 17229, since update 17253, the game constantly crashes after playing one chapter and going back to the main menu
A log file is linked including a session of chapter 1 played on normal difficulty on local play
[RPCS3.log](https://github.com/user-attachments/files/18246959/RPCS3.log)

### Details

The game is composed of chapters that can be played individually, you go from the main menu to chapter scene, after finishing a chatper you go back to the main menu.
Everytime the game tries to load the main menu after succesfully completing the chapter, it crashed
If the chapter is failed, the game behaves as usual.
the crash is reproducible on any chapter with any character as long as the chapter is completed by defeating the boss

Steps to reproduce
- launch a game on any chapter with any character
- win the chapter by defeating the boss
- game crashes on loading menu

### Build with regression

v0.0.34-17253-b2949f11

### Attach two log files

[RPCS3 - Copie.log](https://github.com/user-attachments/files/18246970/RPCS3.-.Copie.log)

### System configuration

The bug has been reproduced on 2 different PCs, one with nvidia RTX 4060 and one nvidia RTX 4070 SUPER

**Comment by AniLeo** [comment]
Invalid regression report, there are more builds than just one between latest good and first bad.

I doubt that the build you blame is the one that introduced the regression, it was probably one of the previous ones.

**Comment by Scuzzlenuts** [comment]
I can corroborate this report, I don't know which build introduced the issue but I had the same problem affecting me with this game. I was on the latest build of RPCS3 and the crashing went away after reverting to 17183 (I just picked a version from a few days prior to having the issue, I'm sure others ahead of it aren't affected).

The last errors in my log files were the same as the submitter's though: "Attempted to register a score that is not better!" which leads me to believe the issue has something to do with the emulator attempting to submit a chapter score to the PS network

**Comment by dcon03** [comment]
@Scuzzlenuts @RocherBrockas Could either of you two test build 17230? I know the last general stable build is 17229, and there is a regression that was introduced in 17230 that affects collection games (working on getting info for a report on that), but it’s still possible that its the build where the regression began.

**Comment by Scuzzlenuts** [comment]
@dcon03 sure! I'll test that out and post a log if it crashes

**Comment by Scuzzlenuts** [comment]
It did indeed crash on 17230

[RPCS3.log](https://github.com/user-attachments/files/18256221/RPCS3.log)

**Comment by dcon03** [comment]
> It did indeed crash on 17230
> 
> [RPCS3.log](https://github.com/user-attachments/files/18256221/RPCS3.log)

17230 is likely the build that broke it then.

Probably edit your message to include a working log from 17229.

**Comment by NomiosTheWanderer** [comment]
Data point of 1, log from a run of Castlevania HD with build 17229.

User Steps:
1.  Launch RPCS3
2. Double click Castlevania HD
3. Press any key to progress past title scsreen
4. Select chapter and choose to begin chapter
5. Complete chapter successfully
6. Victory screen displays, upon proceeding a save occurs
7. Exit to title screen prompting a second save
8. In RPCS3, stop the emulation
9. Close RPCS3

[RPCS3.log](https://github.com/user-attachments/files/18257527/RPCS3.log)

**Comment by NomiosTheWanderer** [comment]
Updated Info:
I have tested this on all builds after the 17229 rollback. There is a different crash occurring at almost the same spot. In the crash described above the failure would occur after the chapter completion stats screen and prior to (possibly during) saving the game. This new crash occurs prior to displaying the stats screen. I also found in testing that the crash does not occur if the network connection (RPCN) is disabled. Here are the various permutations I went through with 17229 and 17257.

[RPCS3 - 17229 - networkEnabled - success.log](https://github.com/user-attachments/files/18268584/RPCS3.-.17229.-.networkEnabled.-.success.log)
[RPCS3 - 17257 - downgraded - networkEnabled - crash.log](https://github.com/user-attachments/files/18268586/RPCS3.-.17257.-.downgraded.-.networkEnabled.-.crash.log)
[RPCS3 - 17257 - freshInstall - networkEnabled - crash.log](https://github.com/user-attachments/files/18268597/RPCS3.-.17257.-.freshInstall.-.networkEnabled.-.crash.log)
[RPCS3 - 17257 - freshInstall - networkDisabled - success.log](https://github.com/user-attachments/files/18268587/RPCS3.-.17257.-.freshInstall.-.networkDisabled.-.success.log)

A possibly relevant note, I could have sworn this was working when I first upgraded to 17257, but sadly I don't have any proof of that happening at this point.
